pytest_asyncio/plugin.py

```diff
@@ -96,21 +103,30 @@ async def async_finalizer():
                         msg = "Async generator fixture didn't stop."
                         msg += "Yield only once."
                         raise ValueError(msg)
-                asyncio.get_event_loop().run_until_complete(async_finalizer())
+                loop.run_until_complete(async_finalizer())
 
             request.addfinalizer(finalizer)
-            return asyncio.get_event_loop().run_until_complete(setup())
+            return loop.run_until_complete(setup())
 
         fixturedef.func = wrapper
     elif inspect.iscoroutinefunction(fixturedef.func):
         coro = fixturedef.func
 
+        strip_event_loop = False
```

Comment:
This code is copy pasted from line 72. Can you refactor it such that it is needed only once? E.g. pull it into a small function.

Reply:
Fixed. Included a new class FixtureStripper. 

Please review again as I have included a new test case, and this has required additional changes
Thanks in advance.

Reply:
Had some minor comments, I am happy overall.